src/instructlab/model/chat.py

```diff
@@ -61,6 +62,30 @@
 
 PROMPT_PREFIX = ">>> "
 
+DEFAULT_ENDPOINT = "http://127.0.0.1:8000/v1"
+
+
+def is_openai_server_and_serving_model(
+    endpoint: str, api_key: str, http_params: HttpClientParams
+) -> bool:
+    """
+    Given an endpoint, returns whether or not the given response is expected
+    from an OpenAI-compatible server, and whether or not it's actively serving
+    a model.
+    """
+    client = OpenAI(
```

Comment:
`chat_cli` already instantiates the client and fetches the models. I think the code should be split into a function and reused. And/or there's `list_models` in `utils.py` that can be of use here instead.

Reply:
It's also worth mentioning that we're only querying a local server for this, so if the server exists then the overhead here would be very minimal. It shouldn't impact the user experience in any way.

Reply:
I am not suggesting you to pass the result of the function called elsewhere. I suggest you reuse the code. It is never simpler to maintain the same code in multiple places. You have a function in `utils.py` that will return what you want. Why not just using it? I am confused. Just call it here and then return `len(...) > 0` as you do.

Reply:
I see - I misread the original comment as suggesting that we reuse the output from the list models function. I will update this to reuse the suggested function.